### DISCUSSION TITLE
Is there anyway to add more items to the database without losing current data?

### DISCUSSION THREAD

**Discussion by rainnfx** [prisma/prisma#24685] [Q&A] [answered]
### :wave: Hey! Prisma beginner here, though I am generally new to using databases on the web.

Curious, after implementing new fields or whatever to `schema.prisma` and running `npx prisma generate` is there a way to push the stuff or whatever without losing data? Whenever I run `npx prisma db push` it warns about data loss and I don't want to lose my data, especially in production!! Is there anyway to get around this? Any explanation would be appreciated! (also sorry if this is a stupid question).

Thanks In advance! :D

**Comment by ludralph** [comment, marked as answer]
Hi @rainnfx 👋 

Thank you for raising this question.

It's not a stupid question at all! Managing schema changes without data loss is a common concern. The `prisma db push` command is designed to quickly synchronize your Prisma schema with your database schema. However, it does not generate migration files and is best suited for prototyping in a development environment. When `prisma db push` detects changes that could result in data loss (e.g., adding a required field without a default value), it will prompt you to reset the database.

You should use [`prisma migrate`](https://www.prisma.io/docs/orm/prisma-migrate/getting-started) commands instead of `prisma db push`. This command helps you create and apply migrations during development. This command helps you create and apply migrations during development. 

- Expand and Contract Pattern:

This pattern allows you to evolve your schema without downtime and data loss. It involves making non-destructive changes first (e.g., adding a nullable field) and then gradually evolving the schema.
You can read more about this pattern in the [documentation](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations#example-use-the-expand-and-contract-pattern-to-evolve-the-schema-without-downtime).

**Reply by rainnfx** [reply]
Thank you so much for the thorough answer! It was what I was hoping for. 👐 
Have a great day! :D
